app/stac_api/views.py

```diff
@@ -491,10 +499,18 @@ def get_queryset(self):
             item__name=self.kwargs['item_name']
         )
 
-    def get_object(self):
+    def check_existence(self):
         asset_name = self.kwargs.get(self.lookup_url_kwarg)
         queryset = self.get_queryset().filter(name=asset_name)
-        obj = get_object_or_404(queryset)
+        return queryset.exists()
+
+    def get_object(self, called_from_within_atomic=False):  # pylint: disable=arguments-differ):
+        asset_name = self.kwargs.get(self.lookup_url_kwarg)
+        queryset = self.get_queryset().filter(name=asset_name)
+        if not called_from_within_atomic:
+            obj = get_object_or_404(queryset)
+        else:
+            obj = Asset.objects.get(name=asset_name)
         return obj
```

Comment:
I believe you can use `get_object_or_404` in an atomic transaction, so imho there's no need to distinguish behaviour here. Am I wrong?

Reply:
Probably you're right. This is just, because I wasn't sure how to handle exceptions within an atomic context, yet. Probably it is ok, if the exceptions are already handled inside the atomic block. But I thought, they need to be handled in a special way. I'll check that